native/test/src/tests/activities/status.rs

```diff
@@ -28,6 +31,75 @@ async fn _setup_accounts(
     Ok(((admin, sync_state1), (observer, sync_state2), room_id))
 }
 
+pub(crate) async fn get_latest_activity(
+    cl: &Client,
+    room_id: String,
+    activity_type: &str,
+) -> Result<Activity> {
+    let retry_strategy = FibonacciBackoff::from_millis(100).map(jitter).take(10);
+    let observer_room_activities = cl.activities_for_room(room_id)?;
+
+    // check the create event
+    let room_activities = observer_room_activities.clone();
+    Retry::spawn(retry_strategy.clone(), move || {
+        let room_activities = room_activities.clone();
+        async move {
+            let stream = room_activities.iter().await?;
+            pin_mut!(stream);
+            let Some(a) = stream
+                .filter(|f| future::ready(f.type_str() == activity_type))
+                .next()
+                .await
+            else {
+                bail!("activity not found")
+            };
+            cl.activity(a.event_meta().event_id.to_string()).await
+        }
+    })
+    .await
+}
+
+#[tokio::test]
+async fn initial_events() -> Result<()> {
+    let _ = env_logger::try_init();
+    let ((admin, _handle1), (observer, _handle2), room_id) =
+        _setup_accounts("room-create-activity").await?;
+
+    // ensure the roomName works on both
+    let activity = get_latest_activity(&admin, room_id.to_string(), "roomName").await?;
+    assert_eq!(activity.type_str(), "roomName");
+
+    let activity = get_latest_activity(&observer, room_id.to_string(), "roomName").await?;
+    assert_eq!(activity.type_str(), "roomName");
+
+    // // check the create event
+    // let room_activities = observer_room_activities.clone();
+    // let created = Retry::spawn(retry_strategy.clone(), move || {
+    //     let room_activities = room_activities.clone();
+    //     async move {
+    //         let Some(a) = room_activities
+    //             .iter()
+    //             .await?
+    //             .find(|f| f.type_str() == "roomCreated")
+    //         else {
+    //             bail!("no create activity found")
+    //         };
+    //         Ok(a)
+    //     }
+    // })
+    // .await?;
+    // assert_eq!(created.type_str(), "created");
+
+    // let ActivityContent::MembershipChange(r) = activity.content() else {
+    //     bail!("not a membership event");}
+    // ;
+    // assert!(matches!(r.change, MembershipChangeType::Invited));
+    // assert_eq!(r.as_str(), "invited");
+    // assert_eq!(r.user_id, to_invite_user_name);
```

Comment:
For reasons not clear to me, I do not receive the `RoomCreate` events, but I wanted to leave the tests code here for future exercise and fixing of the test.